docs/utilities.rst

```diff
@@ -1,25 +1,185 @@
+
+If you want to log at **run-time**, please simply use ``print()`` instead.
+
+.. Note::
+
+    Taichi logging APIs only support standard output now.
+
+.. function:: ti.set_logging_level(level)
+
+    :parameter level: (string) a valid logging level
+
+    - This function is used to set the level of logging. Currently, the log levels in Taichi are sorted as ``ti.TRACE``, ``ti.DEBUG``, ``ti.INFO``, ``ti.WARN`` and ``ti.ERROR``. The default logging level is ``ti.INFO``.
+
+    - The lower the logging level is, the more content will be printed.
+
+        - If we set the logging level to ``ti.TRACE``, all logs will be printed.
+        - If the logging level is ``ti.ERROR``, Taichi shows only those logs generated by ``ti.error()``.
+
+.. Note ::
+
+    You can also override the default logging level by setting the ``TI_LOG_LEVEL`` environment variable. For example,  ``TI_LOG_LEVEL=warn``.
+
+.. function:: ti.info(info)
+
+    :parameter info: (string) logging info
+
+    Print the input string to stdout **only in Taichi-scope**, when the logging level is lower or equal to ``ti.INFO``. For example:
+
+.. code-block:: python
+
+    import taichi as ti
+    ti.init(arch=ti.cpu)
+
+    ti.set_logging_level(ti.INFO)
+    mat = ti.var(dt=ti.f32, shape=(5, 5))
+
+
+    @ti.func
+    def calc(i: ti.int32, j: ti.int32):
+        ti.info(f"set var in ti.func")
+        mat[i, j] = i * j
+
+
+    @ti.kernel
+    def compute():
+        calc(0, 0)
+        calc(1, 1)
+        calc(2, 2)
+        ti.info(f"set var in ti.kernel")
+
+
+    compute()
+    compute()
+    compute()
+
+
+As we statement before, the ``ti.info`` will **only print once** in compile-time. Its output is like:
+
+::
+
+    [I 07/09/20 13:15:24.517] [main.py:calc@10] set var in ti.func
+    [I 07/09/20 13:15:24.518] [main.py:calc@10] set var in ti.func
+    [I 07/09/20 13:15:24.518] [main.py:calc@10] set var in ti.func
+    [I 07/09/20 13:15:24.518] [main.py:compute@19] set var in ti.kernel
+
+The other logging functions below **are all similar to** ``ti.info``. They can all print out when the logging level is set to be lower than they required, respectively.
+
+.. function:: ti.warn(info)
+
+    :parameter info: (string) logging info
+
+.. function:: ti.debug(info)
+
+    :parameter info: (string) logging info
+
+.. function:: ti.trace(info)
+
+    :parameter info: (string) logging info
+
+.. function:: ti.error(info)
+
+    :parameter info: (string) logging info
+
+    This function prints the input string in any logging level and **crashes the program**.
+
+.. warning::
+
+    Note that ``ti.error`` will crash your program and throws an exception ``RuntimeError``.
+
+Here is an example:
+
+.. code-block:: python
+
+    import taichi as ti
+
+    ti.init()
+    ti.set_logging_level(ti.INFO)
+
+    try:
+        ti.error("Fatal error. Exiting now...")
+    except RuntimeError as err:
+        print(err)
+
+
+Profiler
```

Comment:
Profiler should be shared by devs and end-users, maybe we can have a separate .rst for it?

Reply:
IMO I think it's enough in `utilities.rst` at this moment.